spec/validators/ultima_validator_spec.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+require 'rails_helper'
+
+describe UltimaValidator do
+  describe '#validate' do
+    subject(:validator) { described_class.new }
+
+    context 'when batch contains two requests' do
+      let(:record) { create(:batch, request_count: 2) }
+
+      it 'is valid' do
+        validator.validate(record)
+        expect(validator.validate(record)).to be_nil
```

Comment:
This is not a correct assertion. The validation method return value is void (it should not carry meaning for the caller). The assertion needs to be on the errors collection. You could check if record.errors.count is zero or record.errors.full_messages is an empty array. Alternatively, because the error is added on :base, you could check record.errors[:base] is empty.